crates/holochain_conductor_api/src/config/conductor.rs

```diff
@@ -207,7 +185,40 @@ impl ConductorConfig {
 
     /// Check if the config is set to use a rendezvous bootstrap server
     pub fn has_rendezvous_bootstrap(&self) -> bool {
-        self.network.bootstrap_service == Some(url2::url2!("rendezvous:"))
+        self.network.bootstrap_url == url2::url2!("rendezvous:")
+    }
+}
+
+/// All the network config information for the conductor.
+#[derive(Clone, Deserialize, Serialize, Debug, PartialEq, JsonSchema)]
+#[serde(tag = "type", rename_all = "snake_case")]
+pub struct NetworkConfig {
+    /// The Kitsune2 bootstrap server to use for WAN discovery.
+    #[schemars(schema_with = "holochain_util::jsonschema::url2_schema")]
+    bootstrap_url: url2::Url2,
+
+    /// The Kitsune2 sbd server to use for webrtc signalling.
+    #[schemars(schema_with = "holochain_util::jsonschema::url2_schema")]
+    signal_url: url2::Url2,
+
+    /// The Kitsune2 webrtc_config to use for connecting to peers.
+    webrtc_config: Option<serde_json::Value>,
+
+    /// Use this advanced field to directly configure kitsune2.
+    ///
+    /// The above options actually just set specific values in this config.
+    /// Use only if you know what you are doing!
+    advanced: Option<serde_json::Value>,
```

Comment:
For a given version of Holochain, we should know what config options are available in K2. I'd really like to make this strongly typed and be able to reject miss-spelled or out-dated config

Reply:
This is going to be loaded with [set_module_config](https://docs.rs/kitsune2_api/latest/kitsune2_api/struct.Config.html#method.set_module_config) and while it cannot be strongly typed because it depends on which modules are loaded, it *will* be validated by the module validation functions.

I was thinking anything we wanted to strong type like the urls above can be outside this "advanced" config. And this is for everything else.

Reply:
No, it can't be strongly typed, but we can point to the available advanced options in these docs, so that users know what can be configured.

Reply:
Or do you want to have conversions from/to JSON as an in-between layer, @ThetaSinner?

Reply:
Do we just want to delete this entirely until we figure out how we'd like it to look? Or is it okay to keep this for configuring the fiddly advance parameters that most people won't need to use?

Reply:
I'm fine with both as WIP.

Reply:
Likewise, I'm fine with this for now, if you don't mind me having a fiddle later and seeing if I can make this typed :)